content/en/docs/concepts/storage/volumes.md

```diff
@@ -1017,6 +1017,48 @@ spec:
         claimName: my-lamp-site-data
 ```
 
+### Using subPath with expanded environment variables
+
+{{< feature-state for_k8s_version="v1.11" state="alpha" >}}
+
+
+
+You can use expanded environment variables in a subPath value. `subPath` directory names can also
+be constructed from Downward API environment variables. Before you use this feature, you must enable
+the `VolumeSubpathEnvExpansion`feature gate.
+
+ Here is an example of a Pod that uses `subPath` to create a subdirectory within `/logs/` using
+ the pod name, after first creating the environment variable.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: env-image
+spec:
+  containers:
+  - name: env-container
+    env:
+    - name: POD_NAME
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: metadata.name
+    image: busybox
+    command: [ "sh", "-c", "while [ true ]; do echo 'Hello'; sleep 10; done | tee -a /logs/$(POD_NAME)/hello.txt" ]
```

Comment:
I modified this command to actually put the logs where line 1060 claims they go.

Reply:
This is not correct. It should be /logs/hello.txt

Reply:
I don't understand what would intercept the command and make it log to the right place, then? No matter what environment variables are present, I think the command would still log to `/logs/hello.txt`, right? Am I missing something? Does the `subPath` key actually intercept the `tee` command and change its parameters?

Reply:
Yup, the subPath causes hostPath /logs/$(POD_NAME) to be mounted at mountPath "/logs" inside the container.

Maybe this example would be more clear if mountPath and hostPath were different.

Reply:
I agree. Let me see what I can do. As it is right now, it looks a little too "magical".

Reply:
@msau42 Sorry, I can't cut and paste from my example yaml properly. It should have not included the subpath in the `tee`